patches/bnc-onboard++web3-provider-engine+15.0.12.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/node_modules/bnc-onboard/node_modules/web3-provider-engine/index.js b/node_modules/bnc-onboard/node_modules/web3-provider-engine/index.js
+index 3a1b7d5..9bf0f83 100644
+--- a/node_modules/bnc-onboard/node_modules/web3-provider-engine/index.js
++++ b/node_modules/bnc-onboard/node_modules/web3-provider-engine/index.js
+@@ -114,7 +114,11 @@ Web3ProviderEngine.prototype.send = function(payload){
+   throw new Error('Web3ProviderEngine does not support synchronous requests.')
+ }
+ 
+-Web3ProviderEngine.prototype.sendAsync = function(payload, cb){
++Web3ProviderEngine.prototype.sendAsync = function (payload, cb) {
++  if (payload.method === 'eth_getBalance') {
++    return
```

Comment:
Did you test it when transacting? IIRC in some methods, the web3.js calls `getBalance` under the hood, and it can break something

Reply:
Testing it out now across different wallets

Reply:
![image](https://user-images.githubusercontent.com/20442784/153581778-fca477e6-3817-4f4a-bbbe-f64c22d19da8.png)

😢

Reply:
Nice find. It causes issues with the estimation. I've decided to write a modified wallet module instead for WalletConnect that has no balance subscription.